pricegraph/src/orderbook.rs

```diff
@@ -359,20 +359,21 @@ impl Orderbook {
     ///
     /// # Panics
     ///
-    /// This method panics if the provided flow is empty and would not further
-    /// reduce the orderbook. This can happen in certain situations where the
-    /// flow becomes empty due to floating point rounding errors and can cause
-    /// infinite loops when reducing orderbook or enumerating transitive orders.
-    fn fill_path_with_flow(&mut self, path: &[NodeIndex], flow: &Flow) -> Option<()> {
-        if flow.is_empty() {
-            panic!("stuck reducing path with empty flow: {}", format_path(path));
```

Comment:
I think this is the only panic that should be converted into an `Result::Err`. The other ones indicate things that should never happen unless there is:
- a bug
- an incorrect assumption

Reply:
For this panic, it can happen for certain input orderbooks. The other one should not happen for any inputs.

Reply:
I thought that finding a 0 flow (caused by getting an infinite exchange rate) was the root problem.

Reply:
Yes, but this is caused by certain orderbooks when the transitive exchange rate gets too small or too large and becomes either `0` or `∞`. So legitimate and valid orderbooks can cause these FP rounding errors that should propagate a `Result::Err`. However, in the case where a known good path is being filled, but an edge between two tokens can't be found, we should panic as before.

Reply:
We do. I changed the functions to unwrap the get_best_order_for_pair call.

Reply:
:see_no_evil: